snappy.go

```diff
@@ -28,24 +28,37 @@ func Encode(src []byte) []byte {
 // Decode decodes snappy data whether it is traditional unframed
 // or includes the xerial framing format.
 func Decode(src []byte) ([]byte, error) {
+	return DecodeInto(nil, src)
+}
+
+// DecodeInto decodes snappy data whether it is traditional unframed
+// or includes the xerial framing format into the specified `dst`.
+// It is assumed that the entirety of `dst` including all capacity is available
+// for use by this function. If `dst` is nil *or* insufficiently large to hold
+// the decoded `src`, new space will be allocated.
+func DecodeInto(dst, src []byte) ([]byte, error) {
 	var max = len(src)
 	if max < len(xerialHeader) {
 		return nil, ErrMalformed
 	}
 
 	if !bytes.Equal(src[:8], xerialHeader) {
-		return master.Decode(nil, src)
+		return master.Decode(dst[:cap(dst)], src)
```

Comment:
It's alluded to in [the docs](https://godoc.org/github.com/golang/snappy#Decode) but [the code](https://github.com/golang/snappy/blob/master/decode.go#L60) makes it explicit that `len(dst)` is used (not `cap(dst)`, so we make `dst` as big as it gets. (same below for chunk)